src/partition.jl

```diff
@@ -0,0 +1,150 @@
+mutable struct Partition
+    num_districts::Int
+    num_cut_edges::Int
+    total_population::Int
+
+    district_populations::Array{Int, 1} # of length(num_districts)
+    cut_edges::Array{Int, 1}            # of length(num_edges)
+    populations::Array{Int, 1}          # of length(num_nodes)
+    assignments::Array{Int, 1}          # of length(num_nodes)
+    district_adj::SparseMatrixCSC{Int, Int}
+    district_nodes::Array{Set{Int}, 1}
+    attributes::Array{Dict{String, Any}}
```

Comment:
Discussed at meeting: move `attributes`, `total_population`, and `populations` to `Graph` object to avoid copying over and over!